### PR TITLE
Unit tests for Flask app and models

### PR DESCRIPTION
**Pull request by richard-gawne**
Created `tests/test_app.py` containing a set of 11 unit tests for the front-end functionality of `ClinVar-annotator`. All passed.

### CODE REVIEW

**Review by hikhan31 — APPROVED** [review]

**Review by hikhan31 — DISMISSED** [review]

File: tests/test_app.py [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+from unittest.mock import patch
+import pytest
+
+from clinvar_anno_app.app import app, db
+from clinvar_anno_app.models import Patient, Variant, PatientVariant
+
+
+@pytest.fixture
+def client():
+    """Create a test client with a temporary database"""
```
Comment (kaine-veal) [inline tests/test_app.py]: Consider adding more descriptive docstrings throughout, for each test to demonstrate the intent, scope, and expected behaviour of each test case.

File: tests/test_app.py
Comment (kaine-veal) [inline tests/test_app.py:1]: The test suite is well structured and easy to follow, with logical grouping by functionality (routes, uploads, database operations, edge cases).

Excellent use of pytest fixtures to isolate state (temporary database, temp files, reusable mock data).

Just a minor comment to improve docstrings for each test case.

Good work!

File: tests/test_app.py (hunk 1 above)
Comment (hikhan31) [reply tests/test_app.py]: I also agree, it would be great to have more information on what the test does

**Review by hikhan31 — CHANGES_REQUESTED** [review]
Overall everything is looking great. It would be great if you could add some further information to the doc strings.

File: tests/test_app.py
Comment (kaine-veal) [reply tests/test_app.py:1]: I can see you have improved your docstrings for each test case- much more readable. Nice work!

File: tests/test_app.py (hunk 1 above)
Comment (richard-gawne) [reply tests/test_app.py]: Expanded docstring and inline comments
